beacon-chain/sync/data_columns.go

```diff
@@ -99,43 +193,12 @@ func RequestDataColumnSidecarsByRoot(
 				continue
 			}
 
-			// Check if returned data columns align with the block.
-			if err := verify.DataColumnsAlignWithBlock(block, peerSidecars); err != nil {
+			verifiedPeerSidecars, err := verifyColumnsForBlock(block, peerSidecars, newColumnsVerifier)
```

Comment:
Previously, the verification pipeline used to return on error only if `VerifiedRoDataColumn` fail (which should never happen).
In all other cases (for example `Valid` or `SidecarInclusionProven` failed, the sidecar was skipped and considered as no available.

With the new implementation, if one column fail one verification item, then the whole function returns on error.

Reply:
7e39f6358457140fd5ee34d935e96158141ce7cd